grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/BlockingWorker.java

```diff
@@ -0,0 +1,208 @@
+                jedis.del(key(WORKER, name), key(WORKER, name, STARTED), key(STAT, FAILED, name),
+                        key(STAT, PROCESSED, name));
+                return null;
+            }
+        });
+    }
+
+    @Override
+    protected void poll() {
+        // This supports only one queue
+        String curQueue = queueNames.peek();
+        LOG.info("Worker {} consuming from {}", this.name, key(QUEUE, curQueue));
+        while (RUNNING.equals(this.state.get())) {
+            try (Context ignoredPoll = pollTimer.time()) {
+                this.listenerDelegate
+                        .fireEvent(WORKER_POLL, this, curQueue, null, null, null, null);
+                final String payload = pop(curQueue);
+                if (payload != null) {
+                    Job job = ObjectMapperFactory.get().readValue(payload, Job.class);
+                    executor.execute(() -> {
+                        try (Context ignoredProcess = processTimer.time()) {
+                            processingCounter.inc();
+                            process(job, curQueue);
+                            processingCounter.dec();
+                        }
+                    });
+                }
+            } catch (JsonParseException | JsonMappingException e) {
+                // If the job JSON is not deserializable, we never want to submit it again
+                final String fCurQueue = curQueue;
+                executor.execute(() -> PoolUtils.doWorkInPoolNicely(this.jedisPool,
+                        (PoolWork<Jedis, Void>) jedis -> {
+                            removeInFlight(jedis, fCurQueue);
+                            return null;
+                        }));
+                recoverFromException(curQueue, e);
+            } catch (Exception e) {
+                recoverFromException(curQueue, e);
+            }
+        }
+    }
+
+    @Override
+    protected String pop(final String curQueue) {
+        try (Context ignored = popTimer.time()){
+            final String key = key(QUEUE, curQueue);
+            return PoolUtils.doWorkInPoolNicely(this.jedisPool, new PoolWork<Jedis, String>() {
+                /**
+                 * {@inheritDoc}
+                 */
+                @Override
+                public String doWork(final Jedis jedis) {
+                    return jedis.brpoplpush(key, key(INFLIGHT, name, curQueue), BLOCKING_TIMEOUT);
+                }
+            });
+        }
+    }
+
+    private void initWorker() throws IOException {
+        try (Jedis poolResource = jedisPool.getResource()) {
+            poolResource.sadd(key(WORKERS), name);
+            poolResource.set(key(WORKER, name, STARTED), new SimpleDateFormat(DATE_FORMAT).format(new Date()));
+            listenerDelegate.fireEvent(WORKER_START, this, null, null, null, null, null);
+        }
+    }
+
+    @Override
+    protected String lpoplpush(final Jedis jedis, final String from, final String to) {
```

Comment:
By the way, this Blocking worker stuff goes away. It's a hack put there because of how Jesque works. In the next PR I'll just use the library I made